### PR TITLE
Implement "set gaslimit" for KeyManager API

### PR DESCRIPTION
**Pull request by shenhanc**
This PR implements issue #11155 sub task "set gas limit for the validator key".

### What type of PR is this? 

Feature

Implement: Keymanager APIs: get_limit api - subtask: "set gas limt"

### Tested:
1. set gas limit
```
curl --header "Authorization: Bearer ${bearer}" \
  --header "Content-Type: application/json"  \
  --request POST \
  --data '{ "gas_limit": "30000002" }' \
  --write-out "%{http_code}\n" \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```

2. then get
```
curl --header "Authorization: Bearer ${bearer}" \
  --request GET \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```
{"data":{"pubkey":"......","gas_limit":"30000002"}}

### CODE REVIEW

**Comment by james-prysm** [comment]
will review and test early next week, thanks!

**Comment by james-prysm** [comment]
since I looked at this a bit too late you'll need to regenerate that proto will hope to look at it tomorrow

**Comment by shenhanc** [comment]
> since I looked at this a bit too late you'll need to regenerate that proto will hope to look at it tomorrow

Thanks. Synced with develop branch, re-generated the proto files and re-ran tests.

**Review by james-prysm — APPROVED** [review]
Tested using postman and its good, thanks for this contribution 👍

File: validator/rpc/standard_api.go [hunk 1]
```diff
@@ -418,6 +418,64 @@ func (s *Server) GetGasLimit(_ context.Context, req *ethpbservice.PubkeyRequest)
...
+		s.validatorService.ProposerSettings.DefaultConfig != nil &&
+		s.validatorService.ProposerSettings.DefaultConfig.BuilderConfig != nil {
+		// Make a copy of BuildConfig from DefaultConfig (thus "*" then "&"), so when we change GasLimit, we do not mess up with
+		// "DefaultConfig.BuilderConfig".
+		bo := *s.validatorService.ProposerSettings.DefaultConfig.BuilderConfig
+		pBuilderConfig = &bo
+		pBuilderConfig.GasLimit = req.GasLimit
+	} else {
+		// No default BuildConfig to copy from, just create one and set "GasLimit".
+		pBuilderConfig = &validatorServiceConfig.BuilderConfig{GasLimit: req.GasLimit}
```
[the remark this answers is not in the document]
Comment (shenhanc) [reply validator/rpc/standard_api.go]: Thanks. Explicitly added ".. Enabled: False" to newly constructed BuildConfig.

**Comment by shenhanc** [comment]
> Tested using postman and its good, thanks for this contribution +1

Thanks! (Comments addressed by explicitly adding ".. Enabled: False" to newly constructed BuildConfig.)

**Comment by shenhanc** [comment]
Merged PR with the newest develop branch.

**Review by james-prysm — APPROVED** [review]

**Pull request merged**